reporters/kamon-newrelic/src/main/scala/kamon/newrelic/NewRelicConfig.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2013-2020 The Kamon Project <https://kamon.io>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kamon.newrelic
+
+import com.typesafe.config.Config
+import kamon.newrelic.NewRelicConfig.NewRelicApiKey
+import kamon.newrelic.NewRelicConfig.NewRelicApiKey.{InsightsInsertKey, LicenseKey}
+
+import java.net.URL
+import java.time.Duration
+
+
+private case class NewRelicConfig(
+                                   apiKey: NewRelicApiKey,
+                                   enableAuditLogging: Boolean,
+                                   userAgent: String,
+                                   callTimeout: Duration,
+                                   spanIngestUri: Option[URL],
+                                   metricIngestUri: Option[URL]
+                                 )
+
+private object NewRelicConfig {
+
+  def fromConfig(config: Config): NewRelicConfig = {
+    val nrConfig = config.getConfig("kamon.newrelic")
+
+    // TODO maybe some validation around these values?
+    val apiKey = (nrConfig.getString("license-key"), nrConfig.getString("nr-insights-insert-key")) match {
```

Comment:
I have a feeling that if this was modelled as `apiKey: Option[NewRelicApiKey]`, then `none` would not leak into `SpanBatchSenderBuilder`. Right now this `none` needs to be handled both in `SpanBatchSenderBuilder` and `NewRelicConfig`

Reply:
but i have to insert any key into the NR builder `.apiKey(nrConfig.apiKey.value)`.

Reply:
Oh, this is only error log which is logged on missing config, I thought exception is raised. Still don't know why it is even built there if it's known from the start it won't work, but maybe it's not a matter for this PR :P